app/javascript/controllers/auto_citation_controller.js

```diff
@@ -0,0 +1,95 @@
+import { Controller } from "stimulus";
+
+export default class extends Controller {
+  static targets = ["titleField", "contributorFirst", "contributorLast", "contributorRole",
+    "year", "month", "day", "manual", "auto", "switch"];
+
+  connect() {
+    this.purl = this.data.get("purl") || ":link:" // Use a real purl on a persisted item or a placeholder
+
+    this.updateDisplay()
+
+    this.displayDefault(this.switchTarget.checked)
+  }
+
+  updateDisplay() {
+    this.autoTarget.value = this.citation
+  }
+
+  get citation() {
+    return `${this.authorAsSentence} (${this.date}). ${this.title}. Stanford Digital Repository. Available at ${this.purl}`
+  }
+
+  get authorAsSentence() {
+    switch (this.authors.length){
+      case 1:
+      case 2:
+        return this.authors.join(' and ')
+      default:
+        return `${this.authors.slice(0,-1).join(', ')}, and ${this.authors.slice(-1)}`
+    }
+  }
+
+  get authors() {
+    return this.contributorRoleTargets.map((roleField, index) => {
```

Comment:
Does this code currently assume that all contributors are authors?

Reply:
Yes, and I think that's safe to punt on until we get the new designs for authors implemented: https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/420954309

Reply:
That's fine by me as long as, say, it won't barf if the user has one or more org contributors. If that's the case, approving.